### PR TITLE
Add test for simple function using Opaque-based custom type

### PR DESCRIPTION
**Pull request by mbasmanova**
Provide an example of using Opaque-based custom type in a simple function.

### CODE REVIEW

File: velox/expression/tests/CustomTypeTest.cpp [hunk 1]
```diff
@@ -109,13 +125,36 @@ class FromFancyIntFunction : public exec::VectorFunction {
                 .build()};
   }
 };
+
+struct FancyIntT {
+  using type = std::shared_ptr<FancyInt>;
+  static constexpr const char* typeName = "fancy_int";
+};
+using TheFancyInt = CustomType<FancyIntT>;
```
Comment (aditi-pandit) [inline velox/expression/tests/CustomTypeTest.cpp:133]: It looks a bit confusing that we have to create 2 types one derived from CustomType and another from OpaqueType. I would imagine an OpaqueType should be a CustomType. Please can you give some info about it.

Comment (mbasmanova) [reply velox/expression/tests/CustomTypeTest.cpp:133]: @aditi-pandit This is explained in #2965. Also, see #2986 and #2969 .

File: velox/expression/tests/CustomTypeTest.cpp [hunk 2]
```diff
@@ -109,13 +125,36 @@ class FromFancyIntFunction : public exec::VectorFunction {
                 .build()};
   }
 };
+
+struct FancyIntT {
+  using type = std::shared_ptr<FancyInt>;
+  static constexpr const char* typeName = "fancy_int";
```
Comment (xiaoxmeng) [inline velox/expression/tests/CustomTypeTest.cpp:131]:
```
s/static constexpr const char*/static constexpr char*/
```

File: velox/expression/tests/CustomTypeTest.cpp (hunk 1 above)
Comment (xiaoxmeng) [inline velox/expression/tests/CustomTypeTest.cpp:133]:
```
nit: s/TheFancyInt/FancyIntType/
```

**Review by xiaoxmeng — APPROVED** [review]
LGTM. Thanks!

File: velox/expression/tests/CustomTypeTest.cpp (hunk 1 above)
Comment (mbasmanova) [reply velox/expression/tests/CustomTypeTest.cpp:133]: That name is taken. FancyIntType is an instance of velox::Type that describes the type. TheFancyInt is C++ type of the values.

File: velox/expression/tests/CustomTypeTest.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/expression/tests/CustomTypeTest.cpp:131]: This doesn't compile: ISO C++11 does not allow conversion from string literal to 'char *const'

**Pull request closed without merging**